pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -302,13 +318,19 @@ func (r *reconcileContext) doReconcile(ctx context.Context, u *unstructured.Unst
 		}
 	}
 
-	adapter, err := r.Reconciler.model.AdapterForObject(ctx, r.Reconciler.Client, u)
-	if err != nil {
-		if unwrappedErr, ok := lifecyclehandler.CausedByUnresolvableDeps(err); ok {
+	var adapter Adapter
+	var adapteErr error
+	if iamModel, ok := r.Reconciler.model.(IAMModel); ok {
```

Comment:
Personally I think this is more readable if kept in a negative coding style. Not blocking but worth considering.